### PR TITLE
Unit tests not passing on macOS due to inconsistent path resolving

### PR DESCRIPTION
**Pull request by gkedge**
Unit tests not passing on macOS.  autohooks leverage path resolving
whereas the test cases do not.

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -7,7 +7,7 @@ extension-pkg-whitelist=lxml
 
 # Add files or directories to the blacklist. They should be base names, not
 # paths.
-ignore=CVS
+ignore=CVS,.git,.venv,.idea
```
Comment (bjoernricks) [inline .pylintrc:10]: This change is unrelated to the PR. Could you please split it into a separate commit at least?

**Comment by y0urself** [comment]
I never had UnitTest issues on macOS. Can you paste an Output of the failed tests?

**Comment by bjoernricks** [comment]
Could you please also explain why `somepath.resolve()` is required on Mac OS? It should only be required to resolve symlinks.

**Comment by y0urself** [comment]
> Could you please also explain why `somepath.resolve()` is required on Mac OS? It should only be required to resolve symlinks.

It is related to the symbolic link of `/var/` in macOS. It is linked to `/private/var/` in macOS.

**Comment by bjoernricks** [comment]
> > Could you please also explain why `somepath.resolve()` is required on Mac OS? It should only be required to resolve symlinks.
> 
> It is related to the symbolic link of `/var/` in macOS. It is linked to `/private/var/` in macOS.

Thanks for the info. Didn't know this strange thing on Mac OS. In that case the PR looks valid.

**Comment by y0urself** [comment]
Yes. I agree to that.

**Comment by bjoernricks** [comment]
If the commits are cleaned up (see [git rebase -i](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)) we will merge your PR.

**Comment by gkedge** [comment]
> If the commits are cleaned up (see [git rebase -i](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)) we will merge your PR.

I am assuming that you wanted me to do a hard reset on my own forked `master`, commit only the `path.resolve()` changes without the change to `pylintrc` and then forcing that push.  Hope I did the correctly.

If there is a better way, do please share.

**Review by y0urself — APPROVED** [review]

**Comment by bjoernricks** [comment]
> > If the commits are cleaned up (see [git rebase -i](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)) we will merge your PR.
> 
> I am assuming that you wanted me to do a hard reset on my own forked `master`, commit only the `path.resolve()` changes without the change to `pylintrc` and then forcing that push. Hope I did the correctly.
> 
> If there is a better way, do please share.

Exactly. Thanks a lot for your patience and for your contribution!
